modules/gdscript/language_server/gdscript_language_protocol.cpp

```diff
@@ -153,61 +185,45 @@ void GDScriptLanguageProtocol::initialized(const Variant &p_params) {
 }
 
 void GDScriptLanguageProtocol::poll() {
-	server->poll();
+	if (server->is_connection_available()) {
+		on_client_connected();
+	}
+	if (peer != NULL) {
+		StreamPeerTCP::Status status = peer->get_status();
+		if (status == StreamPeerTCP::STATUS_NONE || status == StreamPeerTCP::STATUS_ERROR) {
+			on_client_disconnected();
```

Comment:
Pass disconnected peer to on_client_disconnected for removal from peers container